torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -6049,6 +6049,30 @@ def test_ddp_build_param_to_name_mapping(self):
             param_to_name_mapping = net._build_param_to_name_mapping(net_params)
             self.assertDictEqual(expected_mapping, param_to_name_mapping)
 
+        def test_ddp_build_param_to_name_mapping_requires_grad(self):
+            class Net(nn.Module):
+                def __init__(self):
+                    super().__init__()
+                    self.lin = nn.Linear(10, 10)
+                    # Is not tracked by DDP and should not show up in param to
+                    # name mapping.
+                    self.lin.bias.requires_grad_(False)
```

Comment:
Shall we add another test that validate the `ValueError` if `requires_grad_` is false?

Reply:
That ValueError is kind of internal to DDP, it would be a bug in DDP if user could construct inputs that trigger that error. For example, that error would be triggered before this requires_grad fix. Will add a line to those errors that it is a bug and to report an issue.

Reply:
Actually, I guess we can test these error messages by manually changing parameters in this test case and above, so might as well add that.